sample-apps/README.md

```diff
@@ -38,3 +38,245 @@ List of sample apps across all repositories in [aws-observability](https://githu
+
+**Pros of adding invocation pattern**
+
+* Better demonstration of context propagation
+* Sample apps across different languages can call each other which better represents a distributed system
+* Is an optional configuration 
+
+**Cons of adding invocation pattern**
+
+* Configuring each sample app to call each other is not as straight forward as just having a traffic generator
+* Another layer of configurability is added
+
+Sample apps in conformance to this document will also include the invocation pattern as an optional configuration. This will allow for more extensive testing and demos if configured.
+A possible example can be seen below. This invocation capability can allow testing to be as complex or simple as needed.
+
+![Alt text](./invocation.png?raw=true "Invocation")
+
+**Metrics** can be more loosely generated if they are not related to incoming requests. The two patterns of metrics that will be generated can be categorized as **Request-based** and **Random**. Request based metrics will also assume an invocation. Random can be arbitrarily generated for metrics that do not correlate to incoming requests. 
+
+### Documentation Requirements
+
+Each sample app MUST be coupled with a document that will have the following sections:
+
+* **Description** - Information about the language being used and a link to this document for context
+* **Non-conformance** - Documentation on missing features or missing extensions that don’t allow a given sample app to conform with this document
+* **Workarounds** - If a given sample app is using any workarounds to conform to this document (such as using unstable sdk & api OR writing in the variable names) then the sections of the code that could break in the future must be documented
+* **README** - The readme that will go into each sample app’s code base detailing how to setup the sample app (this should be same across all sample apps); although redundant, each repository should still have it to avoid looking across sample apps for a readme. This readme should also include the Non-conformance section specific to the sample app it belongs to.
+
+### Functional Requirements
+
+In a perfect world where everything is stable upstream, every sample app should emit the same Metrics, Logs, and Traces if given a specific routine (e.g. Python and Go sample apps both receive the same 3 requests → both emit the same signals in the same format). The following specifications will detail what exactly will be tracked as a metric log or trace. Each sample app should be configured with manual instrumentation from OTEL. Auto instrumentation can also be implemented.
+Metrics and traces should be OTLP and Logs should be in JSON format using any popular logging library for each respective language with the appropriate license. (e.g. Log4j for java, Logrus for Go)
```

Comment:
```suggestion
Metrics and traces should be OTLP and Logs should be in JSON format using any popular logging library for each respective language with the appropriate license. (e.g. Log4j for java, Logrus for Go)
```
Do any of the already built sample apps generate logs? Should we remove reference to them in this doc until they are stable?

Reply:
I just checked some of the apps, and I do not think they generate logs yet.  Would it be a pain to re-add logs into the doc later?